eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/RetryHttpSinkHandler.java

```diff
@@ -0,0 +1,214 @@
+                    // add the data to the queue
+                    ((WebhookHttpSinkHandler) sinkHandler).addDataToQueue(exportRecord);
+                }
+            })
+            .onRetry(e -> {
+                if (log.isDebugEnabled()) {
+                    log.warn("Retrying the request to {} for the {} time. HttpConnectRecord= {}", url, e.getAttemptCount(), httpConnectRecord);
+                } else {
+                    log.warn("Retrying the request to {} for the {} time.", url, e.getAttemptCount());
+                }
+                if (connectorConfig.getWebhookConfig().isActivate()) {
+                    HttpExportRecord exportRecord = covertToExportRecord(e, e.getLastResult(), e.getLastException(), url, firstTryId);
+                    ((WebhookHttpSinkHandler) sinkHandler).addDataToQueue(exportRecord);
+                }
+            })
+            .onFailure(e -> {
+                if (log.isDebugEnabled()) {
+                    log.error("Failed to send the request to {} after {} attempts. HttpConnectRecord= {}", url, e.getAttemptCount(),
+                        httpConnectRecord, e.getException());
+                } else {
+                    log.error("Failed to send the request to {} after {} attempts.", url, e.getAttemptCount(), e.getException());
+                }
+                if (connectorConfig.getWebhookConfig().isActivate()) {
+                    HttpExportRecord exportRecord = covertToExportRecord(e, e.getResult(), e.getException(), url, firstTryId);
+                    ((WebhookHttpSinkHandler) sinkHandler).addDataToQueue(exportRecord);
+                }
+            }).build();
+
+        // Handle the HttpConnectRecord with retry
+        Failsafe.with(retryPolicy)
+            .getStageAsync(() -> sinkHandler.deliver(url, httpConnectRecord).toCompletionStage());
+
+        return null;
+    }
+
+    /**
+     * Converts the ExecutionCompletedEvent to an HttpExportRecord.
+     *
+     * @param event      the ExecutionCompletedEvent to convert
+     * @param response   the response of the request, may be null
+     * @param e          the exception thrown during the request, may be null
+     * @param url        the URL the request was sent to
+     * @param firstTryId the UUID of the first try
+     * @return the converted HttpExportRecord
+     */
+    private HttpExportRecord covertToExportRecord(ExecutionEvent event, HttpResponse<Buffer> response, Throwable e, URI url, String firstTryId) {
+        HttpExportMetadataBuilder builder = HttpExportMetadata.builder()
+            .url(url.toString())
+            .receivedTime(LocalDateTime.now())
+            .retryNum(event.getAttemptCount() - 1);
+
+        if (event.getAttemptCount() == 1) {
+            builder.retriedBy(null)
+                .uuid(firstTryId);
+        } else {
+            builder.retriedBy(firstTryId)
+                .uuid(UUID.randomUUID().toString());
+        }
```

Comment:
I think using UUIDs is a great thing, but in this PR, UUIDs are only used in the `/export` endpoint. If you prefer a combination of `connector ip:port + topic + offset` (which does work as a unique ID), you can replace the UUID with this unique ID; if you prefer a UUID, then the UUID can be used more broadly to help the user keep track of “response received from callbackUrl” versus “response sent to callbackUrl”.

Reply:
> If you prefer a combination of connector ip:port + topic + offset (which does work as a unique ID), you can replace the UUID with this unique ID;

I believe UUID is a better choice. I'm not certain if `topic` and `offset` are always non-null and unique. If we can't ensure their existence and uniqueness, then `connector ip:port + topic + offset` can't guarantee uniqueness either.

> if you prefer a UUID, then the UUID can be used more broadly to help the user keep track of “response received from callbackUrl” versus “response sent to callbackUrl” (When the Sink Connector receives a response from callbackUrl, the UUID used to send the request is added to the metadata of the response.

What I understand is: you want to add a unique id for the callback data as well, like in the JSON structure below:

```json
{
  "type": "httpSink.http.common",
  "timestamp":"1714382902219",
  "uuid": "fefaa2be-fe46-4024-8f44-8f97b3add35d", 
  "data": "callbackData"
}
```
and the JSON structure for users to obtain callback data is as follows:
```json
{
  "pageNum": 1,
  "pageSize": 10,
  "pageItems": [
    {
      "metadata": {
        "url": "http://example.com",
        "code": 500,
        "receivedTime": "2024-04-28T14:21:31.091",
        "uuid": "second-delivery-uuid",
        "retriedBy": "first-delivery-uuid",
        "retryNum": 1
      },
      "data": {
        "type": "httpSink.http.common",
        "timestamp": "1714382902219",
        "uuid": "fefaa2be-fe46-4024-8f44-8f97b3add35d",
        "data": "callbackData"
      }
    }
  ]
}
```

Reply:
>I believe UUID is a better choice.

Ok, it is true that the offset of the retried event is the same when retrying. Then you can use the UUID as the unique key for the cast in the Sink Connector (since the HTTP Sink Connector has a retry mechanism that uses the UUID for management) and keep the offset as the EventMesh's unique ID for the event.

>What I understand is: you want to add a unique id for the callback data as well, like in the JSON structure below:

Add a offset unique ID field:

```json
{
  "type": "httpSink.http.common",
  "timestamp":"1714382902219",
  "uuid": "fefaa2be-fe46-4024-8f44-8f97b3add35d", 
  "eventId": "testTopic-37564"
  "data": "callbackData"
}
```

>and the JSON structure for users to obtain callback data is as follows:

>When the Sink Connector receives a response from callbackUrl, the UUID used to send the request is added to the **metadata** of the response.

Put UUID in the **metadata** instead of data:

```json
{
  "pageNum": 1,
  "pageSize": 10,
  "pageItems": [
    {
      "metadata": {
        "url": "http://example.com",
        "code": 500,
        "receivedTime": "2024-04-28T14:21:31.091",
        "uuid": "the-uuid-of-this-retry",
        "retriedBy": "fefaa2be-fe46-4024-8f44-8f97b3add35d",
        "retryNum": 1
      },
      "data": ${HttpExportRecord.data}
    },
    {
      "metadata": {
        "url": "http://example.com",
        "code": 500,
        "receivedTime": "2024-04-28T14:21:30.091",
        "uuid": "fefaa2be-fe46-4024-8f44-8f97b3add35d",
        "retriedBy": null,
        "retryNum": null
      },
      "data": ${HttpExportRecord.data}
    }
  ]
}
```

The request sent and the response received will share the same UUID as the delivery ID. This shouldn't lead to conflict, right?

Reply:
pubSubConfig.subject + offset or more like connector info

Reply:
Resolved at https://github.com/apache/eventmesh/pull/4837/commits/4af7455c9b09d4a1771fa7580b7406a61160ea74